### ISSUE TITLE
Bug 572427 - Decreased performance of DependencyComputation during initial class path resolution 

### ISSUE THREAD

**Issue by HannesWell** [eclipse/tycho#47]
See https://bugs.eclipse.org/bugs/show_bug.cgi?id=572427

Current state:
The performance problems of the `DependencyComputer` is fixed in Tycho by caching the `org.eclipse.osgi.container.ModuleWires` of each `org.eclipse.osgi.container.ModuleWiring`. The root of the problem has been reported to the [Equinox-Framework bug-tracker](https://bugs.eclipse.org/bugs/show_bug.cgi?id=572605) and a fix is under development.
So if everything works as expected and the performance improvements meet the expectations the [fix in Tycho](d8d4a0c694b0a6785e7c779c91ff8ae0bbedabd8) can be reverted once the improved version of `org.eclipse.osgi` is available for Tycho.

**Comment by HannesWell** [comment]
With PR #161 one big step to improve the dependency resolution runtime is now fully completed.
The dependency resolution now only takes about two times longer than with Tycho `2.2.0`, compared to about six time the runtime from Tycho 2.2.0 to 2.3.0.

As already mentioned in the PR, in my benchmark the runtimes are the followings (all on the same computer with the same build and only different tycho versions):
```
From Scanning for projects to to long line before the reactor summary
Tycho 2.2.0 - 25sec (from which 7sec are Scanning for projects)
Tycho 2.3.0 - 147sec (from which 7sec are Scanning for projects)
Tycho 2.4.0-SNAPSHOT - 48sec (from which 7sec are Scanning for projects)
```
The following arguments were used to display a timestamp next to each printed line from which the runtime was derived:
`-Dorg.slf4j.simpleLogger.showDateTime=true -Dorg.slf4j.simpleLogger.dateTimeFormat=HH:mm:ss`

These runtimes and their ratio of course depend on the exact build.
However since the runtime is still about two times greater than before and other issues (like #159) regarding the runtime were raised, I keep this open and will have a look later if even better runtime can be achieved.

Other builds for benchmarking would be useful.

**Comment by HannesWell** [comment]
After issue #159 is closed with PR #160 the runtime of the dependency resolution is dropped to 32sec (including scanning for projects which still takes 7 sec).

**Comment by HannesWell** [comment]
In my latest measurements the Dependency-Resolution part only took about 6 or 7 sec in my case. So I think there is not much more to achieve in this regard.
